xdsl/irdl/declarative_assembly_format.py

```diff
@@ -35,24 +35,21 @@ class ParsingState:
     It contains the elements that have already been parsed.
     """
 
-    operands: list[UnresolvedOperand | None]
-    operand_types: list[Attribute | None]
-    result_types: list[Attribute | None]
+    operands: list[UnresolvedOperand | None | list[UnresolvedOperand | None]]
+    operand_types: list[Attribute | None | list[Attribute | None]]
+    result_types: list[Attribute | None | list[Attribute | None]]
     attributes: dict[str, Attribute]
     constraint_variables: dict[str, Attribute]
 
     def __init__(self, op_def: OpDef):
-        if op_def.attributes or op_def.regions or op_def.successors:
+        attributes = bool(op_def.attributes) and list(op_def.attributes.keys()) != [
+            "operandSegmentSizes"
+        ]
+        if attributes or op_def.regions or op_def.successors:
```

Comment:
`attributes` threw me off here, I feel like `has_attributes` communicates that it's a `bool`
```suggestion
        has_attributes = bool(op_def.attributes) and list(op_def.attributes.keys()) != [
            "operandSegmentSizes"
        ]
        has_if attributes or op_def.regions or op_def.successors:
```